### PR TITLE
GrabCut Dashboard

### PR DESCRIPTION
**Pull request by jlstevens**
This PR demonstrates how the interactive visualizations developed and used in the notebooks can be made to work as dashboards served using bokeh server. It also demonstrates how a workflow can be made to work in stages using multiple browser tabs.

Stage 1:

image

Stage 2 (based on the region selected in stage 1):

image

(Obviously this is just showing that grab cut works as this is not a useful contour!)

The third stage (development in progress) will be to mesh the chosen contour into a trimesh.

Currently, my biggest obstacle for these first two stages is that I cannot slice the selected tile returned by quest (due to what look like incorrect coordinates) as described in https://github.com/pyviz/EarthSim/pull/174.

### CODE REVIEW

**Comment by kcpevey** [comment]
Need to add bounds to `minimum_size = param.Integer(default=10, bounds=(0, 1000))` (line 145 in grabcut.py).  Re: https://github.com/pyviz/panel/issues/33

**Comment by kcpevey** [comment]
This error from above `TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType` happens when the zoom level is a really low number for an image that is highly zoomed in.

**Comment by jlstevens** [comment]
@kcpevey Thanks for reporting these issues! I trust that these errors appear in those particular edge cases (e.g that bbox and when zoomed in a lot) and is otherwise (mostly) working?

**Comment by kcpevey** [comment]
@jlstevens I'm not seeing any other errors, but I do suspect there is another tricky bug in there somewhere. Occasionally the `update contour` button doesn't work (nothing appears on the right). I know that it can take a lot of time with no feedback, but this is NOT that. Basically, the `update contour` button randomly decides to stop working no matter how long I wait. *Usually* if I pan around a bit, mess with the min size slider, and click the `filter contour` button a few times and maybe throw in one more `update contour` for kicks, then it will run properly again and polygons will appear on the right.   
  
In summary - I think there's a bug, but its so elusive I haven't been able to even begin to track it down.

**Comment by kcpevey** [comment]
Also, a minor suggestion - I'd like the ability to instantiate the grabcut dashboard with drawn bbox. I'm working to have the workflows set up to be able to be run as-is if the user would just like to check things out. This is particularly important with the grabcuts tool because finding a land/water region which is appropriate can take effort (e.g.  coloration difference between land/water is significant)

**Comment by kcpevey** [comment]
It seems like the zoom level that's being selected is lower than what I anticipate. At one point, I drew a bbox and the resulting image was literally 6 *giant* pixels. I messed around a bit and redrew roughly the same box and I got a reasonable image (see below). I can't make the giant pixel (poor zoom level selection) happen again, but I did see something similar happen the other day as well. 
grabcut

**Comment by kcpevey** [comment]
I apologize for my vague reporting, I just can't seem to track anything down. Strange things are happening and I'm not sure why, but I will continue to try and figure it out and report back any findings. An example of strange behavior is this image below. I see no reason why grabcut doesn't pick up the left river bank.  
grabcut2

**Comment by kcpevey** [comment]
Just did this:  
grabcut5a  
And got this:  
grabcut5b
  
Then after running a cell with this:
```
rgb = grabcut.GrabCutDashboard.tiff_to_rgb(tiff_path)
grabcuts = grabcut.GrabCutDashboard(rgb, fg_data=[], bg_data=[], height=600,
                                              name='GrabCut Extractor', minimum_size=200)
panel.Column(grabcuts, grabcuts.view())
```
I immediately double checked `selection.zoom_level` and it had reset itself to 14 (which is what it should have been). So there is some issue with the zoom level not being set at the correct time.

**Comment by philippjfr** [comment]
I'll close in favor of #253 but won't delete the branch in case there's anything we want to reuse from this dashboard in the future.
